### PR TITLE
[FIX] web: fix calendar and datetime picker layout

### PR DESCRIPTION
**Pull request by chgo-odoo**
[FIX] web: fix calendar and datetime picker layout

=== ISSUE 1 ===

The spacing between the navigation buttons in Calendar were managed
by a `gap-1` class. This worked but it was not consistent with
the navigation we have in project gantt view. In order to improve
the consistency of our  UI, we need to make this look the same.

=== ISSUE 2 ===

If you navigate to Time Off and request a new time off, the datetime
picker `apply` position is not correct. This is due to the fact that
no `state.timeValues` is visible to position it accordingly. That means
we need to specify two different layout depending if these buttons are
visible or not.

== AFTER ===

We remove the `gap-1` class and add a `btn-group` class instead,
which manage the spacing between children buttons.

We also change the navigation icons to `arrows` one and fix two issues
related to the datetime picker : 

- The gap between month name isn't computed as it should ;
- There is an issue with the position of the `apply` button
when no `state.timeValues` is displayed which is fixed by specifying two different layout depending if `state.timeValues` is visible or not.

task-3389679
part of task-3326263

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/datetime/datetime_picker.scss [hunk 1]
```diff
@@ -138,7 +138,8 @@
     // Compute  spacing  between the two month by compensating buttons size and tables gap
     
     .o_zoom_out {
-        gap: o-to-rem($btn-border-width) * 4 + o-to-rem($o-font-size-base) * 2 + o-to-rem($btn-padding-x * 4);
+
```
Comment (Brieuc-brd) [inline addons/web/static/src/core/datetime/datetime_picker.scss:141]: Useless empty space

**Comment by chgo-odoo** [comment]
Hi @Brieuc-brd 👋

Could you check the changes to this PR, please?

Thanks

File: addons/web/static/src/core/datetime/datetime_picker.xml [hunk 2]
```diff
@@ -48,8 +48,8 @@
                 </div>
             </t>
         </div>
-        <div class="d-flex flex-column flex-md-row justify-content-center position-relative gap-4 gap-md-3">
-            <div class="d-flex gap-3 w-100 justify-content-center">
+        <div t-attf-class="d-flex flex-column flex-md-row position-relative gap-4 gap-md-3 {{ props.type === 'datetime' ? 'justify-content-center' : 'justify-content-end' }}">
```
Comment (Brieuc-brd) [inline addons/web/static/src/core/datetime/datetime_picker.xml:51]: Can you put `position-relative` at the beginning?

**Review by Brieuc-brd — COMMENTED** [review]
@chgo-odoo Thanks for your work 👍 
Just a little comment below.

Also, since you changed the calendar controller AND the datetime_picker, could you adapt your commit message and your PR title/ description ?

**Comment by chgo-odoo** [comment]
@Brieuc-brd Commit/PR title & description have been adapted to fit the changes of this PR. Runbot is 🟢, could you check?

Thanks !

File: addons/web/static/src/core/datetime/datetime_picker.scss [hunk 3]
```diff
@@ -138,7 +138,7 @@
     // Compute  spacing  between the two month by compensating buttons size and tables gap
     
     .o_zoom_out {
-        gap: o-to-rem($btn-border-width) * 4 + o-to-rem($o-font-size-base) * 2 + o-to-rem($btn-padding-x * 4);
+        gap: calc(#{$btn-border-width * 4} +  #{$o-font-size-base * 2} +  #{$btn-padding-x * 4});
```
Comment (Brieuc-brd) [inline addons/web/static/src/core/datetime/datetime_picker.scss:141]: I don't think it's necessary to use `calc` in this case. I think the main issue comes from adding values to `o-to-rem` that are already in `rem`, which divides them by 16 and creates the gap problem.
Could you try this?
```suggestion
        gap: o-to-rem($btn-border-width) * 4 + $o-font-size-base * 2 + $btn-padding-x * 4;
```

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Hey @chgo-odoo 👋 Thanks for your work :)
A little comment below.

File: addons/web/static/src/core/datetime/datetime_picker.scss (hunk 3 above)
Comment (chgo-odoo) [reply addons/web/static/src/core/datetime/datetime_picker.scss]: That doesn't work either, we already tried to find a workaround with converting only the first value to rem (I still pushed on runbot so you can check). IMO we should go for a calc if it avoids all these potential issues

**Review by Brieuc-brd — APPROVED** [review]
@chgo-odoo LGTM 👍  you can remove the draft state

Hey @pparidans 👋 This PR is ready for tech. review :)

**Review by pparidans — APPROVED** [review]
Changes LGTM
@chgo-odoo could you rebase your branch to fix conflicts that appeared since your last push ?

Thx

**Comment by chgo-odoo** [comment]
Hi @pparidans 👋 **Don't** r+ this one yet please, I rebased but I need to fix something, thanks !

**Comment by pparidans** [comment]
> Hi @pparidans wave **Don't** r+ this one yet please, I rebased but I need to fix something, thanks !

No problem. Ping me when you think it's OK and I'll do a second round of review.

**Comment by chgo-odoo** [comment]
Good morning @pparidans 👋

This one is finally ready for review, I needed to fix a small error inside the computation of the gap.

Could you check? Thanks !

**Pull request closed without merging**